test-app/test-uv.js

```diff
@@ -1,6 +1,12 @@
 var p = nucleus.dofile("deps/utils.js").prettyPrint;
 var uv = nucleus.uv;
 
+function assert(condition, message) {
+    if (!condition) {
+        throw message || "Assertion failed";
```

Comment:
Should this be `throw new Error(message || "Assertion failed")` so that JS can show a stack trace better?
I don't know about duktape, but other engines don't show stack traces always if you throw plain strings.